### PR TITLE
Added private newsletter webview option for a newsroom fixes #110

### LINKED ISSUE #110: Private switch for newsroom
Add a toggle-able switch for the newsroom that will require a user to login before they can view the content on the newsroom

### PR DESCRIPTION
**Pull request by sarongebre**
### _April 10, 2024_ 

I think it's good to go for a review. I would like to mention that Sonar Cloud does not like some of the existing code in the enews_sample_sata.sql, enews.sql, and upgrade.php files. Should I refactor those?

### _April 8, 2024_
First version of the code. Still testing to make sure everything works right.

I can change some of the labels and/or descriptions for the newsletter private web view option. 

When we add a new column to the database, what's the procedure for updating the main server's database? Should I attach an SQL file to the issue, containing the ALTER TABLE SQL commands needed to make the changes?

I can create some functions like 'getNewsroom' and 'Authenticate' in the UNL_ENews_Newsletter_Public file that can be referenced by the constructor in that file if it will help clean up some code.

closes #110

### CODE REVIEW

**Comment by sarongebre** [comment]
### _April 10 2024_ 

I think it's good to go for a review. I would like to mention that Solar Cloud does not like some of the existing code in the enews_sample_sata.sql, enews.sql, and upgrade.php files. Should I refactor those?

### _April 8 2024_
First version of the code. Still testing to make sure everything works right.

I can change some of the labels and/or descriptions for the newsletter private web view option. 

When we add a new column to the database, what's the procedure for updating the main server's database? Should I attach an SQL file to the issue, containing the ALTER TABLE SQL commands needed to make the changes?

I can create some functions like 'getNewsroom' and 'Authenticate' in the UNL_ENews_Newsletter_Public file that can be referenced by the constructor in that file if it will help clean up some code.

File: src/UNL/ENews/Newsletter/Story.php [hunk 1]
```diff
@@ -70,6 +70,12 @@ function __construct($options = array())
             if (!($this->newsroom->shortname == $options['shortname'])) {
                 throw new Exception('Not a valid newsroom name.', 400);
             }
+
+            if(isset($this->newsroom->private_web_view) && $this->newsroom->private_web_view === '1' ) {
```
Comment (ericras) [inline src/UNL/ENews/Newsletter/Story.php:74]: formatting

Comment (sarongebre) [reply src/UNL/ENews/Newsletter/Story.php]: Added a space after "if". Is that what you meant with the formatting? Thanks.

File: src/UNL/ENews/Manager.php [hunk 2]
```diff
@@ -70,6 +70,14 @@ function handlePost()
                 } else {
                     $_POST['allow_submissions'] = 0;
                 }
+
+                if (isset($_POST['private_web_view'])
+                && $_POST['private_web_view'] == 'on') {
+                $_POST['private_web_view'] = 1;
```
[the remark this answers is not in the document]
Comment (sarongebre) [reply src/UNL/ENews/Manager.php]: Changed the separated "if" condition to all be in one line.

**Pull request merged**